### PR TITLE
chore: separate tag configs

### PR DESCRIPTION
**Pull request by hikchoi**
# chore: separate tag configs
This PR:
- Splits up `enableFrontmatterTags` and `enableHashesForFMTags` to be in `preview` and `publishing` namespace
  - originally (before publishing namespace migration), both preview and publishing behavior was set by one config under `site.{enableFrontmatterTag|enableHashesForFMTags}`

No diff in madge output

### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [x] sticking to existing conventions instead of creating new ones 
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended

- General
  - [x] check whether code be simplified
  - [x] check if similar function already exist in the codebase. if so, can it be re-used?
  - [x] check if this change adversely impact performance
- Operations
  - [x] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [~] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed 
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics

- [~] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### Extended

- [~] can we track the performance of this change to know if it is _successful_? 
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

## 

### Basics

- [x] [Write Tests](dev.process.qa.test) 
- [x] [Confirm existing tests pass](dev.process.qa.test)
- [x] [Confirm manual testing](dev.process.qa.test) 
- [x] Common cases tested
- [x] 1-2 Edge cases tested
- [x] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

### Extended

- [x] If you are adding a new language feature (graphically visible in VS Code/preview/publishing), an example is included in the [test workspace](dev.ref.test-workspace)

- CSS
  - [~] display is correct for following dimensions
    - [~] sm: screen ≥ 576px, eg. iphonex, (375x812)
    - [~] lg: screen ≥ 992px
    - [~] xxl: screen ≥ 1600px eg. mac (1600x900)
  - [~] display is correct for following browsers (across the various dimensions)
    - [~] safari
    - [~] firefox
    - [~] chrome

### Basics

- [x] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR

## 

### Basics

- [~] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](dev) or [Packages](pkg)

## 

### Extended

- [~]  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](dev.process.close-loop)
  - eg. [breaking dev change due to new dependency](https://github.com/dendronhq/dendron/pull/2188#pullrequestreview-855696330)

### CODE REVIEW

File: packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap [hunk 1]
```diff
@@ -294,6 +333,72 @@ VFile {
...
+`;
+
+exports[`dendronPub frontmatter tags GIVEN enableFrontmatterTags: true THEN rendered when available: single tag, PUBLISHING 2`] = `
+VFile {
+  "contents": "<h1 id=\\"fmtags\\"><a aria-hidden=\\"true\\" class=\\"anchor-heading\\" href=\\"#fmtags\\"><svg aria-hidden=\\"true\\" viewBox=\\"0 0 16 16\\"><use xlink:href=\\"#svg-link\\"></use></svg></a>Fmtags</h1>
+<p>has fm tags</p>
+<hr>
+<h2 id=\\"tags\\"><a aria-hidden=\\"true\\" class=\\"anchor-heading\\" href=\\"#tags\\"><svg aria-hidden=\\"true\\" viewBox=\\"0 0 16 16\\"><use xlink:href=\\"#svg-link\\"></use></svg></a>Tags</h2>
+<ol>
+<li><a title=\\"Private\\" style=\\"color: brown\\" href=\\"https://wiki.dendron.so/notes/hfyvYGJZQiUwQaaxQO27q.html\\" target=\\"_blank\\">first (Private)</a></li>
```
Comment (SeriousBug) [inline packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap:359]: Is the fact that these links are rendered private unrelated to this change? The snap before the PR has them rendered non-private.

Comment (hikchoi) [reply packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap:359]: Not sure why the snap before PR had them as non-private, but in the test `THEN rendered when available: multiple tags: publishing`, I only create the note `has.fmtags`, and the tags that this note has are `first and `second`. Since both `tags.first` and `tags.second` don't exist, I think they _should_ be private?

Comment (hikchoi) [reply packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap:359]: Just checked. all proc tests that have `ProcFlavor.PUBLISHING` are correctly rendering them to be private since I don't create the tag notes in any of the tests.

All test cases for `ProcFlavor.PREVIEW` won't have them rendered as private because we only do that for publishing.

Comment (hikchoi) [reply packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap:359]: > Not sure why the snap before PR had them as non-private, 

Looks like before the PR, the proc that was used to test didn't pass in the proc flavor as publishing, thus making it default to `ProcFlavor.REGULAR`.

I've confirmed that rolling back to before the PR, passing in `ProcFlavor.PUBLISHING` to the proc, and running it as-is will update the snapshot so that it correctly renders tags `first` and `second` as private.

**Comment by hikchoi** [comment]
hmm not sure why CI is failing. it's not coming from any files that touched. is it caching the packages?

**Comment by kevinslin** [comment]
> hmm not sure why CI is failing. it's not coming from any files that touched. is it caching the packages?

looks like a compiler issue in Gotonote: https://github.com/dendronhq/dendron/runs/5303577105?check_suite_focus=true#step:9:49

unused variable import

**Comment by hikchoi** [comment]
> > hmm not sure why CI is failing. it's not coming from any files that touched. is it caching the packages?
> 
> looks like a compiler issue in Gotonote: https://github.com/dendronhq/dendron/runs/5303577105?check_suite_focus=true#step:9:49
> 
> unused variable import

I'm aware of that. Just that I am not touching that file and what the CI is complaining isn't present.
(see: https://github.com/dendronhq/dendron/blob/chore/separate-tag-configs/packages/plugin-core/src/commands/GotoNote.ts#L27:L27)

**Comment by hikchoi** [comment]
The tests are failing because an unused import was re-introduced during a merge in https://github.com/dendronhq/dendron/commit/8d359d39c54672fdebbd647cd8bff1867b6000d2

On this branch, I rebased just be for the merge mentioned above happened, so on my end it looked like `GoToNote.ts` didn't have any problem and I was fine building locally.

Suspecting that CI is caching the artifacts from another run before which is affecting subsequent runs. (though the cache action is saying that they didn't find any cached libs, so I'm a bit confused.)

**Comment by hikchoi** [comment]
rebased and checks are all passing now

**Review by kevinslin — APPROVED** [review]
